cfgrammar/src/lib/yacc/grammar.rs

```diff
@@ -89,8 +89,8 @@ pub struct YaccGrammar<StorageT = u32> {
     actions: Vec<Option<String>>,
     /// The programs section of a grammar, if specified; otherwise `None`.
     programs: Option<String>,
-    /// The %actiontype declaration value, if specified; otherwise `None`.
-    actiontype: Option<String>,
+    /// The actiontypes of rules (one per rule).
+    actiontypes: Vec<Option<String>>,
```

Comment:
Why is `String` optional? Don't all rules need to have a type if we are using actions?

Reply:
Ah, or is this backward compatibility for the original YaccKind.

Reply:
Yes, but some grammar types (e.g. parse trees) don't need any actions. It turns out to be easier to deal with `Vec<Option<String>>` than `Option<Vec<String>>`.

Reply:
Ok.